### ISSUE TITLE
Mattermost Personal Token (Admin) NOT shown AT ALL in 4.5

### ISSUE THREAD

**Issue by limenleap** [mattermost/mattermost-server#8064]
#### Summary
Mattermost Personal Token (Admin) NOT shown AT ALL in 4.5 This is serious issue for me as I have an external application that uses the API and the personal token for admin to do some work
#### Steps to reproduce
Login as admin. Go to the "Account Settings" Click on Security. Create a personal token. It will pop up the warning saying it is for the admin . .. you still proceed. Then nothing at all happens

#### Expected behavior
One is expected to see the personal token ONCE. (and later on it is not to be shown)

#### Observed behavior (that appears unintentional) 
It simply does not show anything. In fact that dialog does not seem to get populated. Only when you close the dialog, and repeat the steps of opening the Security dialog and then go to the personal access token area and the new token is indicated there. **but we cannot copy the actual token, as it is now anyway hidden**

#### Possible fixes
Not sure if this is due to browser. I checked the browser inspector console and did not see any message. Checked it in Firefox and Chrome.  The issue is still present

**Comment by lindy65** [comment]
Hi @limenleap,

Thanks for your feedback,

I have tried to reproduce your issue on v4.6 but am not able to.

For expected behavior, please see [this documentation](https://docs.mattermost.com/developer/personal-access-tokens.html).

Version 4.6 ships on January 16th.  If you'd like to test on a server running this version you can use [this server](http://rctesting.reddogsofwar.com/)

**Comment by limenleap** [comment]
Hi. I rechecked umpteen times but the issue is surely there. My server has only 1GB RAM... can that be one reason? Not sure. The hosting is on Amazon Lightsail on Ubuntu 16.04 -- can that be having some peculiarities? Again, not sure

I will try and make a video/gif of what I see when I get time for this

**Comment by limenleap** [comment]
@lindy65 I tried the server link you gave but unfortunately the Personal Access Token feature is not switched on for me. I tried creating a team called "abcdefg" there but sorry, this feature was not available on that server (it must be switched off in the system console/config file) so I can't test

**Comment by limenleap** [comment]
@lindy65 I think I got a clue: The SMTP settings in my installation is possibly not working (Will debug that separately) So I think I am getting this issue because Mattermost is unable to send a notification mail to me saying that a personal token has been created

If that be the case; I think Mattermost developers should change the flow of the logic. I think the personal token should be shown the way it was done before ... and the problem of not being able to send that notification email should be handled separately and not be tied to the display of the private token (which is done only once)

Hope this helps

**Comment by lindy65** [comment]
Hi @limenleap,

Apologies for the delay in responding, our team was testing / investigating your issue...

Could you share a screenshot of what you see after you hit "Yes, Create" in the popup that warns about creating a System Admin access token?

**Comment by limenleap** [comment]
Hi it comes back to the earlier dialog as if I had not even pressed the "Save" button. More I think about this more it looks that it is due to the fact that my SMTP server config has gone wrong somewhere

(I am travelling and so may not be able to respond till Monday 12, Jan)

Image of the third step:
![step3-admintoken](https://user-images.githubusercontent.com/20351066/34769378-a3180014-f623-11e7-843d-e72f50ddd380.jpg)
Image of the second step:

![step2-admintoken](https://user-images.githubusercontent.com/20351066/34769501-00241c48-f624-11e7-939f-dd11fa4667c6.jpg)

Image of the first step:

![step1-admintoken](https://user-images.githubusercontent.com/20351066/34769519-0942a272-f624-11e7-96e5-ddd4eefa03ae.jpg)

**Comment by limenleap** [comment]
Also, when I close the entire Account Settings dialog, then come back again I find that the security token indeed has been created! But I can't copy the token now anyway. I get only the token ID and not the token itself

![securitytokencreated](https://user-images.githubusercontent.com/20351066/34769673-7051b44e-f624-11e7-8d96-cd9c2a51ab56.jpg)

**Comment by jasonblais** [comment]
Thanks @limenleap! 

If you hit "Save" again on the third screen, does it do the same?

I've tried to reproduce but I'm unable to. One other thing that would be useful is getting a snapshot of your JavaScript console while you're creating the token. Here is how for various browsers: https://webmasters.stackexchange.com/questions/8525/how-do-i-open-the-javascript-console-in-different-browsers

**Comment by limenleap** [comment]
Yes it does create yet another one. And again the same problem happens.. then again I hit save and one more and on and on iteratively. 

Also, I am not getting any email saying that the token is created. Though it has. Because when I close the account settings and start up the security link etc I do see the list of tokens created in the aforesaid fashion

I suspect something is going wrong when the email is not being sent to me. Possibly you should start there

No the JavaScript console does not show anything. I am right now travelling so I can't check again but I remember checking for clues in the JavaScript console and there was nothing there 

Sabu Francis
Chief Punner, Limen Leap Labs
Limen Leap Labs is a division of Sabu Francis & Associates, Navi Mumbai, India
+918976016428 (Outside Ooty, India) +919488482833 (Ooty, India)
Skype: sabu.francis Twitter: @sabufrancis, @limenleap, @teamtad Telepathy: ~readmymind

Notice:
If you are not the intended recipient, any dissemination, distribution, copying of this email or even laughing about it over a couple of beers or whatever is not authorized (either explicitly or implicitly) and constitutes an irritating social faux pas. So please notify me quietly -- for which I will be eternally grateful. Then just delete this email immediately, and pretend it never happened

---- On Wed, 10 Jan 2018 19:51:04 +0530 notifications@github.com wrote ----

Thanks @limenleap!

If you hit "Save" again on the third screen, does it do the same?

I've tried to reproduce but I'm unable to. One other thing that would be useful is getting a snapshot of your JavaScript console while you're creating the token. Here is how for various browsers: https://webmasters.stackexchange.com/questions/8525/how-do-i-open-the-javascript-console-in-different-browsers

**Comment by jasonblais** [comment]
@limenleap Yeah, I tried with different email settings, including one where the configuration was incorrect.

Maybe when you're back later this week you can help share your email settings from config.json? You can hide any confidential information of course.

**Comment by lindy65** [comment]
Hi @limenleap,

Wondering whether you've managed to solve your issue?

**Comment by limenleap** [comment]
Sorry. Still out of station. I will be back on Feb 6. Last I heard from office it wasn't resolved. That server on Lightsail has been deleted. Will try again and report back

---- On Wed, 24 Jan 2018 14:24:38 +0530 notifications@github.com wrote ----

Hi @limenleap,

Wondering whether you've managed to solve your issue?

**Comment by lindy65** [comment]
No problem, we'll catch up once you're back at work...

**Comment by lindy65** [comment]
Hi @limenleap 

Have you managed to resolve this issue yet?

**Comment by lindy65** [comment]
Hi @limenleap,

Have you perhaps had time to share your email settings from config.json?

**Comment by lindy65** [comment]
Hi @limenleap,

Popping in to let you know that [v4.8.0](https://about.mattermost.com/download/) released today so perhaps you'd like to upgrade to see if this issue still persists?
